src/util/sandbox.cpp

```diff
@@ -369,80 +369,119 @@ QString Sandbox::migrateOldSettings() {
     // Otherwise, developers would need to run with --settingsPath every time or symlink
     // to use the same settings directory with signed and unsigned builds.
 
-    // QDir::homePath returns a path inside the sandbox.
+    // QDir::homePath returns a path inside the sandbox when running sandboxed
     QString homePath = QLatin1String("/Users/") + qgetenv("USER");
```

Comment:
We cannot rely on userdata unsanitized.  Can you add a check with `qEnvironmentVariableIsEmpty`? We should also check if there are any slashes in the name.

```
$ env | grep USER=
USER=jan
$ export USER=
$ env | grep USER=
USER=
```

Reply:
I don't really understand your point. Usernames can't have slashes in them or that would probably cause all kinds of havok on any Unix system. In an extreme edge case where someone has bizarrely overriden the USER environment variable manually (I've never heard of anyone doing this and I don't think there's any reason to ever do so), this is going to fail regardless.

Reply:
Yeah, and in that case we should display an error message and exit.